plugins/jira-dashboard-backend/src/api.ts

```diff
@@ -93,14 +94,15 @@ export type SearchOptions = {
  * @param config - A Backstage config
  * @param jqlQuery - A string containing the jql query.
  * @param options - Query options that will be passed on to the POST request.
- *
+ * @returns A promise that resolves with the search results and status code.
+ * @throws If an error occurs during the search process.
  * @public
  */
 export const searchJira = async (
   config: Config,
   jqlQuery: string,
   options: SearchOptions,
-): Promise<Issue[]> => {
+): Promise<SearchJiraResponse> => {
```

Comment:
Can we change the error handling a bit? I would prefer if this would return "JiraQueryResults" directly. What is the benefit of having the caller of this function have to check the response status? Maybe something like this?

```ts
  const response = await fetch(`${resolveJiraBaseUrl(config)}search`, {
  ....
  }); 
  if (response.ok) {
     return await response.json();
  }
  throw new Error(`${response.status}`)

```